### ISSUE TITLE
duplicate_argument_linter should not lint `abort()` calls

### ISSUE THREAD

**Issue by olivroy** [r-lib/lintr#2565]
```r
cli::cli_inform(c(
  "i" = "xx",
  "i" = "yy"
))
```
Line 3[duplicate_argument_linter] Avoid duplicate arguments in function calls.

named vectors with names "!", "i", "*", "x", ">" maybe should not be linted as they are bullet names?

https://cli.r-lib.org/reference/cli_bullets.html#details for the list of bullets

More generally, sometimes, we can use:

```r
msg <- c(
  i = "x",
  i = "y",
  "*" = "z"
)
cli::cli_abort(msg)
```

Maybe the solution would just be to add more cases to the default.

```r
except = c("mutate", "transmute", "abort", "warn", "inform", "cli_abort", "cli_warn", "cli_inform")
```

**Comment by MichaelChirico** [comment]
the exception here is the call to c(), not the rlang calls. c() is intentionally included by default; you should add "c" as an exception for this case. don't think there's anything else lintr should do here.
